include/zephyr/net/phy.h

```diff
@@ -73,7 +73,7 @@ typedef void (*phy_callback_t)(const struct device *dev,
  * These are for internal use only, so skip these in
  * public documentation.
  */
-__subsystem struct ethphy_driver_api {
```

Comment:
Dunno, the keyword seems to be used only by `parse_syscalls.py` and `gen_kobject_list.py` scripts. So it looked like it is not needed here.

Reply:
Would love the quick explanation of what `__subsystem` does myself, and what's the impact for the drivers that (inadvertently?) are missing it?

```console
$ grep "driver_api {" . -R | grep -v "__subsystem"

./usb_c/usbc_vbus.h:struct usbc_vbus_driver_api {
./espi_emul.h:struct emul_espi_driver_api {
./interrupt_controller/dw_ace.h:struct dw_ace_v1_ictl_driver_api {
./interrupt_controller/intel_vtd.h:struct vtd_driver_api {
./display.h:struct display_driver_api {
./video.h:struct video_driver_api {
./i3c/target_device.h:struct i3c_target_driver_api {
./clock_control.h:struct clock_control_driver_api {
./pcie/endpoint/pcie_ep.h:struct pcie_ep_driver_api {
./lora.h:struct lora_driver_api {
./led_strip.h:struct led_strip_driver_api {
./i2c.h:struct i2c_target_driver_api {
```

Reply:
Partly answering myself: guess this is how the build system knows it should look for functions in that struct that might be flagged with `__syscall`? So some of the drivers mentioned above might need updating to allow using them from user mode?

Reply:
Adding `__subsystem` does more than syscall processing. It generates means for validating the type of this particular kernel object (it generates `K_OBJ_DRIVER_ETHPHY`).

I'd like to see all driver APIs using `__subsystem`, even the ones not exposing any syscalls, as this allows for more rigorous checks - whether used in-tree or not.

Reply:
> I'd like to see all driver APIs using `__subsystem`, even the ones not exposing any syscalls, as this allows for more rigorous checks - whether used in-tree or not.

I wonder if `__subsystem` would also help with documentation aspects of driver APIs?
Right now many drivers (including ETH PHY here) are hiding the API struct from public documentation, but it might be better to expose it, but with some kind of visual cue that it's mainly useful to driver developers. Related: #63286.